blocks/bricks/sequences.py

```diff
@@ -107,11 +112,15 @@ class MLP(Sequence, Initializable, Feedforward):
 
     """
     @lazy(allocation=['dims'])
-    def __init__(self, activations, dims, **kwargs):
+    def __init__(self, activations, dims, prototype=None, **kwargs):
         self.activations = activations
-
-        self.linear_transformations = [Linear(name='linear_{}'.format(i))
-                                       for i in range(len(activations))]
+        self.prototype = Linear() if prototype is None else prototype
+        self.linear_transformations = []
+        for i in range(len(activations)):
+            linear = copy.deepcopy(self.prototype)
```

Comment:
@bartvm For reference, 

```
>>> a = Linear()
>>> b = copy.deepcopy(a)
>>> a.apply.application == b.apply.application
True
```

So I think whatever Dima did in #974 handles deepcopy correctly.